docs/api-diffs/9.0/rc.1/diff.md

```diff
@@ -0,0 +1,1290 @@
++            public HealthStatus? HealthStatus { get; set; }
++            public DateTime? StartTimeStamp { get; set; }
++            public DateTime? StopTimeStamp { get; set; }
++            public ImmutableArray<VolumeSnapshot> Volumes { get; set; }
+         }
++        public sealed class DeploymentTargetAnnotation : IResourceAnnotation {
++            public DeploymentTargetAnnotation(IResource target);
++            public IResource DeploymentTarget { get; }
++        }
+         [DebuggerDisplayAttribute("Type = {GetType().Name,nq}, Name = {Name}")]
+         public sealed class EndpointAnnotation : IResourceAnnotation {
+-            public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, string? transport = null, string? name = null, int? port = default(int?), int? targetPort = default(int?), bool? isExternal = default(bool?), bool isProxied = true);
++            public EndpointAnnotation(ProtocolType protocol, string? uriScheme = null, string? transport = null, [EndpointNameAttribute] string? name = null, int? port = default(int?), int? targetPort = default(int?), bool? isExternal = default(bool?), bool isProxied = true);
++            public string TargetHost { get; set; }
+         }
++        [AttributeUsageAttribute(2048, AllowMultiple=false)]
++        public class EndpointNameAttribute : Attribute, IModelNameParameter {
++            public EndpointNameAttribute();
++        }
+-        public class ExecutableResource : Resource, IResource, IResourceWithArgs, IResourceWithEndpoints, IResourceWithEnvironment
++        public class ExecutableResource : Resource, IResource, IResourceWithArgs, IResourceWithEndpoints, IResourceWithEnvironment, IResourceWithWaitSupport
++        public class ExecuteCommandContext {
++            public ExecuteCommandContext();
++            public required CancellationToken CancellationToken { get; set; }
++            public required string ResourceName { get; set; }
++            public required IServiceProvider ServiceProvider { get; set; }
++        }
++        public class ExecuteCommandResult {
++            public ExecuteCommandResult();
++            public string ErrorMessage { get; set; }
++            public required bool Success { get; set; }
++        }
++        [DebuggerDisplayAttribute("Type = {GetType().Name,nq}, Key = {Key}")]
++        public class HealthCheckAnnotation : IResourceAnnotation {
++            public HealthCheckAnnotation(string key);
++            public string Key { get; }
++        }
++        public sealed class HealthReportSnapshot : IEquatable<HealthReportSnapshot> {
++            public HealthReportSnapshot(string Name, HealthStatus Status, string Description, string ExceptionText);
++            public string Description { get; set; }
++            public string ExceptionText { get; set; }
++            public string Name { get; set; }
++            public HealthStatus Status { get; set; }
++        }
++        public enum IconVariant {
```

Comment:
@JamesNK - is "variant" the best word/term to use here? Would `IconState` be better?

Reply:
Variant is what FluentUI is uses. Since the icon is based on a fluentui name, we should stick with their terminology.